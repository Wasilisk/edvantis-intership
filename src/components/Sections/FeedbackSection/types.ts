import {FeedbackType} from "../../../models";

export type FeedbackSectionProps = {
    feedbacks: FeedbackType[]
}