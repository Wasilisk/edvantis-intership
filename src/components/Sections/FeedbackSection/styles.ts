import styled from "styled-components";
import {FeedbackCardStyles} from "../../Cards/FeedbackCard/styles";

export const FeedbackSectionStyles = styled.div`
  background-color: #f5f5f5;
  padding: 0 24px;
  
  .wrapper {
    max-width: 1200px;
    margin: 0 auto;
    padding: 70px 0;
    display: flex;
    flex-direction: column;
    align-items: center;

    .title {
      font-size: 38px;
      color: #333333;
      line-height: 1.2;
      font-weight: 600;
      margin: 0;
      text-align: center;
    }

    .description {
      font-size: 18px;
      line-height: 1.56;
      font-weight: 400;
      color: #333333;
      margin: 25px 0 0 0;
      text-align: center;
    }
    
    .feedback-cards {
      display: flex;
      justify-content: space-between;
      align-items: stretch;
      margin-top: 50px;
      
      & > ${FeedbackCardStyles} {
        margin: 0 15px;
      }
    }
  }
  
  @media(max-width: 768px) {
    .wrapper {
      .feedback-cards {
        & > ${FeedbackCardStyles}:last-child {
          display: none;
        }
      }
    }
  }
  @media(max-width: 600px) {
    .wrapper {
      .feedback-cards {
        flex-wrap: wrap;
        & > ${FeedbackCardStyles}:first-child {
          margin-bottom: 15px;
        }
      }
    }
  }
  @media(max-width: 425px) {
    .wrapper {
      .feedback-cards {
        & > ${FeedbackCardStyles}:not(:first-child) {
          display: none;
        }
        & > ${FeedbackCardStyles}:first-child {
          margin-bottom: 0;
        }
      }
    }
  }
`;