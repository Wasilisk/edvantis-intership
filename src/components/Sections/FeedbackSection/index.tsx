import React from 'react';
import { FeedbackSectionStyles } from './styles';
import {FeedbackSectionProps} from "./types";
import FeedbackCard from "../../Cards/FeedbackCard";

const FeedbackSection = ({feedbacks}: FeedbackSectionProps) => {
    return (
        <FeedbackSectionStyles>
            <div className="wrapper">
                <h4 className="title">What Our Customers Say</h4>
                <p className="description">We love your feedback on our products!</p>
                <div className="feedback-cards">
                    {
                        feedbacks.slice(0, 3).map(feedback => <FeedbackCard key={feedback.id} {...feedback}/>)
                    }
                </div>
            </div>
        </FeedbackSectionStyles>
    );
};

export default FeedbackSection;