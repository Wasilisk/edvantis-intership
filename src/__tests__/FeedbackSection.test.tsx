import {render} from "./__utils__/store-wrapper";
import FeedbackSection from "../components/Sections/FeedbackSection";
import {mockFeedback} from "./__mock__/mock-feedback";

describe('FeedbackSection component', () => {
    it('should render header component', () => {
        const {asFragment, getByText, getAllByTestId} = render(<FeedbackSection
            feedbacks={[...Array(4)].map((_, index) => Object.assign({}, mockFeedback, { id: index }))}
        />);

        const title = getByText('What Our Customers Say');
        expect(title).toBeDefined();

        const description = getByText('We love your feedback on our products!')
        expect(description).toBeDefined();

        const feedbacks = getAllByTestId('feedback-card');
        expect(feedbacks.length).toBe(3);

        expect(asFragment()).toMatchSnapshot();
    })
})