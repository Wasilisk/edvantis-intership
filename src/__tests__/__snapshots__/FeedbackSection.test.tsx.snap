// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackSection component should render header component 1`] = `
<DocumentFragment>
  <div
    class="sc-gsnTZi iFuclJ"
  >
    <div
      class="wrapper"
    >
      <h4
        class="title"
      >
        What Our Customers Say
      </h4>
      <p
        class="description"
      >
        We love your feedback on our products!
      </p>
      <div
        class="feedback-cards"
      >
        <div
          class="sc-bczRLJ iKlFUT"
          data-testid="feedback-card"
        >
          <p
            class="username"
          >
            Jim Mockeen
          </p>
          <p
            class="department"
          >
            About SOFA - 40 Warren St, Boston
          </p>
          <div
            class="rating"
          >
            <div
              class="sc-dkzDqf iYDomr"
            >
              <div
                class="sc-hKMtZM iikMon"
              >
                <div
                  class="filled"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
                <div
                  class="empty"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
              </div>
              <div
                class="sc-hKMtZM iikMon"
              >
                <div
                  class="filled"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
                <div
                  class="empty"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
              </div>
              <div
                class="sc-hKMtZM iikMon"
              >
                <div
                  class="filled"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
                <div
                  class="empty"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
              </div>
              <div
                class="sc-hKMtZM iikMon"
              >
                <div
                  class="filled"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
                <div
                  class="empty"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
              </div>
              <div
                class="sc-hKMtZM iikMon"
              >
                <div
                  class="filled"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
                <div
                  class="empty"
                >
                  <svg>
                    empty-star.svg
                  </svg>
                </div>
              </div>
            </div>
            <span>
              8/10
            </span>
          </div>
          <p
            class="comment"
          >
            I not only found the perfect sofa at a reasonable price but also stylish lamps which were exactly what I wanted. Thank you, SOFA!
          </p>
          <p>
            August 07, 2022
          </p>
        </div>
        <div
          class="sc-bczRLJ iKlFUT"
          data-testid="feedback-card"
        >
          <p
            class="username"
          >
            Jim Mockeen
          </p>
          <p
            class="department"
          >
            About SOFA - 40 Warren St, Boston
          </p>
          <div
            class="rating"
          >
            <div
              class="sc-dkzDqf iYDomr"
            >
              <div
                class="sc-hKMtZM iikMon"
              >
                <div
                  class="filled"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
                <div
                  class="empty"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
              </div>
              <div
                class="sc-hKMtZM iikMon"
              >
                <div
                  class="filled"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
                <div
                  class="empty"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
              </div>
              <div
                class="sc-hKMtZM iikMon"
              >
                <div
                  class="filled"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
                <div
                  class="empty"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
              </div>
              <div
                class="sc-hKMtZM iikMon"
              >
                <div
                  class="filled"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
                <div
                  class="empty"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
              </div>
              <div
                class="sc-hKMtZM iikMon"
              >
                <div
                  class="filled"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
                <div
                  class="empty"
                >
                  <svg>
                    empty-star.svg
                  </svg>
                </div>
              </div>
            </div>
            <span>
              8/10
            </span>
          </div>
          <p
            class="comment"
          >
            I not only found the perfect sofa at a reasonable price but also stylish lamps which were exactly what I wanted. Thank you, SOFA!
          </p>
          <p>
            August 07, 2022
          </p>
        </div>
        <div
          class="sc-bczRLJ iKlFUT"
          data-testid="feedback-card"
        >
          <p
            class="username"
          >
            Jim Mockeen
          </p>
          <p
            class="department"
          >
            About SOFA - 40 Warren St, Boston
          </p>
          <div
            class="rating"
          >
            <div
              class="sc-dkzDqf iYDomr"
            >
              <div
                class="sc-hKMtZM iikMon"
              >
                <div
                  class="filled"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
                <div
                  class="empty"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
              </div>
              <div
                class="sc-hKMtZM iikMon"
              >
                <div
                  class="filled"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
                <div
                  class="empty"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
              </div>
              <div
                class="sc-hKMtZM iikMon"
              >
                <div
                  class="filled"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
                <div
                  class="empty"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
              </div>
              <div
                class="sc-hKMtZM iikMon"
              >
                <div
                  class="filled"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
                <div
                  class="empty"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
              </div>
              <div
                class="sc-hKMtZM iikMon"
              >
                <div
                  class="filled"
                >
                  <svg>
                    filled-star.svg
                  </svg>
                </div>
                <div
                  class="empty"
                >
                  <svg>
                    empty-star.svg
                  </svg>
                </div>
              </div>
            </div>
            <span>
              8/10
            </span>
          </div>
          <p
            class="comment"
          >
            I not only found the perfect sofa at a reasonable price but also stylish lamps which were exactly what I wanted. Thank you, SOFA!
          </p>
          <p>
            August 07, 2022
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
